kDrive/AppRouter.swift

```diff
@@ -0,0 +1,440 @@
+                                                message: KDriveResourcesStrings.Localizable.modalDeletePhotosDescription,
+                                                action: KDriveResourcesStrings.Localizable.buttonDelete,
+                                                destructive: true,
+                                                loading: false) {
+                Task {
+                    // Proceed with removal
+                    @InjectService var photoCleaner: PhotoLibraryCleanerServiceable
+                    await photoCleaner.removePicturesScheduledForDeletion()
+                }
+            }
+
+            window?.rootViewController?.present(alert, animated: true)
+        }
+    }
+
+    public func askForReview() async {
+        guard let presentingViewController = await window?.rootViewController,
+              !Bundle.main.isRunningInTestFlight else {
+            return
+        }
+
+        guard reviewManager.shouldRequestReview() else {
+            return
+        }
+
+        let appName = Bundle.main.object(forInfoDictionaryKey: "CFBundleName") as! String
+
+        Task { @MainActor in
+            let alert = AlertTextViewController(
+                title: appName,
+                message: KDriveResourcesStrings.Localizable.reviewAlertTitle,
+                action: KDriveResourcesStrings.Localizable.buttonYes,
+                hasCancelButton: true,
+                cancelString: KDriveResourcesStrings.Localizable.buttonNo,
+                handler: requestAppStoreReview,
+                cancelHandler: openUserReport
+            )
+
+            presentingViewController.present(alert, animated: true)
+        }
+        MatomoUtils.track(eventWithCategory: .appReview, name: "alertPresented")
+    }
+
+    @MainActor private func requestAppStoreReview() {
+        MatomoUtils.track(eventWithCategory: .appReview, name: "like")
+        UserDefaults.shared.appReview = .readyForReview
+        reviewManager.requestReview()
+    }
+
+    @MainActor private func openUserReport() {
+        MatomoUtils.track(eventWithCategory: .appReview, name: "dislike")
+        guard let url = URL(string: KDriveResourcesStrings.Localizable.urlUserReportiOS),
+              let presentingViewController = window?.rootViewController else {
+            return
+        }
+        UserDefaults.shared.appReview = .feedback
+        presentingViewController.present(SFSafariViewController(url: url), animated: true)
+    }
+
+    public func refreshCacheScanLibraryAndUpload(preload: Bool, isSwitching: Bool) async {
+        Log.appDelegate("refreshCacheScanLibraryAndUpload preload:\(preload) isSwitching:\(preload)")
+
+        availableOfflineManager.updateAvailableOfflineFiles(status: ReachabilityListener.instance.currentStatus)
+
+        do {
+            // TODO: accountManager migrate to Transactionable (not tracking live objects)
+            await Task { @MainActor in
```

Comment:
You cannot await a Task like this. 
You have to first create a reference to the task:
`let someTask =  Task { @MainActor in`
And later
`try await someTask.result`

Else the code will just run past this block and also ignore try catch.

Reply:
If this whole block must run on MainActor you can maybe move it in a separate function annotated @MainActor

Reply:
will do 👍